url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/model3d/TriangleNet3DComp.java

```diff
@@ -0,0 +1,2604 @@
+				query.remove(elts[i]);
+
+				Point3D po = null;
+				Iterator<Equivalentable> it;
+
+				/*
+				 * If the neighbours already exist, delete them from the query
+				 * set, so that they are not tested again.
+				 */
+				TriangleElt3D te = null;
+				te = elts[i].getNeighbour(0);
+				if (te != null)
+					query.remove(te);
+				te = elts[i].getNeighbour(1);
+				if (te != null)
+					query.remove(te);
+				te = elts[i].getNeighbour(2);
+				if (te != null)
+					query.remove(te);
+
+				for (int m = 0; m < 3; m++) {
+					int p1 = 0;
+					int p2 = 0;
+					switch (m) {
+					// check at what neighbour's points it should be
+					case 0:
+						p1 = 1;
+						p2 = 2;
+						break;
+					case 1:
+						p1 = 0;
+						p2 = 2;
+						break;
+					case 2:
+						p1 = 0;
+						p2 = 1;
+					}
+
+					if ((elts[i].getNeighbour(m)) == null) {
+						po = elts[i].getPoint(p1);
+						it = query.iterator();
+
+						while (it.hasNext()) {
+							TriangleElt3D oTE = (TriangleElt3D) it.next();
+
+							for (int j = 0; j < 3; j++) {
+								Point3D ot = oTE.getPoint(j);
+								if (po.isEqual(ot, so)) {
+									// a point is equal
+									Point3D po1 = elts[i].getPoint(p2);
+
+									for (int k = 0; k < 3; k++) {
+										if (k != j) {
+											Point3D ot1 = oTE.getPoint(k);
+											if (po1.isEqual(ot1, so)) {
+												// 2nd point => they are
+												// neighbours
+
+												elts[i].setNeighbour(m, oTE);
+												// neighbour registered at this
+												int otind;
+												// index for other TriangleElt3D
+
+												if (j != 2 && k != 2)
+													// old comparison
+													// (j==0
+													// &&
+													// k==1)
+													// ||
+													// (j==1&&k==0))
+													otind = 2;
+												// this is oppossite of point 2
+												else if (j != 0 && k != 0)
+													// old comparision
+													// (j==1
+													// &&
+													// k==2)
+													// ||
+													// (j==2
+													// &&
+													// k==1))
+													otind = 0;
+												else
+													otind = 1;
+
+												oTE.setNeighbour(otind, elts[i]);
+												it.remove();
+												break;
+											}
+										}
+									}
+									break;
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks net topology.
+	 * 
+	 * @param elts
+	 *            TriangleElt3D for which the topology should be checked
+	 * @throws IllegalArgumentException
+	 *             if the index of the point of the tetrahedron is not in the
+	 *             interval [0;3]. The exception originates in the method
+	 *             getPoint(int) of the class Tetrahedron3D.
+	 * @throws IllegalArgumentException
+	 *             - if index of a triangle point is not 0, 1 or 2. The
+	 *             exception originates in the method getPoint(int) of the class
+	 *             Triangle3D.
+	 */
+	public void checkNetTopology(TriangleElt3D[] elts) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 37. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.